x/staking/keeper/msg_server.go

```diff
@@ -84,7 +84,7 @@ func (k msgServer) ProposalUnjailValidator(goCtx context.Context, msg *types.Msg
 		return nil, err
 	}
 
-	if !validator.IsJailed() {
+	if !validator.IsJailed() { // TODO: check other statuses Active | Inactive | Paused | Jailed
```

Comment:
When unjailed, status should be Inactive so that user can activate it. It start counting for activation time from inactive status and activate it.

Reply:
We need have a proposal to jail a validator by governance